File: scripts/verify.js
const fs = require("fs");
const path = require("path");

const ROOT = path.resolve(__dirname, "..");

function fail(message) {
  console.log(`FAIL: ${message}`);
  process.exit(1);
}

function ensureFile(name) {
  const filePath = path.join(ROOT, name);
  if (!fs.existsSync(filePath)) {
    fail(`missing ${name}`);
  }
}

ensureFile("index.html");
ensureFile("game.js");
ensureFile("AGENTS.md");

const htmlPath = path.join(ROOT, "index.html");
const html = fs.readFileSync(htmlPath, "utf8");
const hasCanvas = /<canvas[^>]*id=["']game["'][^>]*>/i.test(html);
if (!hasCanvas) {
  fail('index.html missing canvas id="game"');
}

console.log("PASS");

File: scripts/verify.py
import os, sys

ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

def fail(msg):
    print("FAIL:", msg)
    sys.exit(1)

def ok(msg):
    print("OK:", msg)

required = ["index.html", "game.js", "AGENTS.md"]
for f in required:
    p = os.path.join(ROOT, f)
    if not os.path.exists(p):
        fail(f"missing {f}")
    ok(f"found {f}")

print("PASS")
